ui/packages/consul-ui/app/components/code-editor/index.js

```diff
@@ -35,6 +35,9 @@ export default Component.extend({
       ...DEFAULTS,
       mode: mode.mime,
       readOnly: this.readonly,
+      htmlMode: mode.htmlMode,
+      matchClosing: mode.matchClosing,
+      alignCDATA: mode.alignCDATA,
```

Comment:
The only little quibble I have here, is that fact that only the XML mode has these options on mode I think? Not sure if it matters at all, what do you think?

Should we check the mode and only add these options if you are in XML mode? It's been a long time since I've been in this code myself, so not massively sure if it matters (probably not) or if we should do it just through "correctness" (or maybe on the contrary "this is fine, lets leave as it")

Lemme know what you think, pretty sure whichever way we go we can merge, wondered what you thought?

Reply:
Yeah, only XML mode has those options. Initially, I thought it was ok since the other modes don't have them. But I'll make some changes to make this clearer in the code.